### ISSUE TITLE
GK requests to GC

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#4806]
Looking at the debug logs I saved yesterday (2015.03.29-NB) I have now seen requests like this 
`http://www.geocaching.com/track/details.aspx?tracker=GKA301`
in the c:geo log in parallel to the correct request to geokrety.org.

Today 2015.03.30-NB1 I cannot reproduce this anymore.
@kumy Was that solved by recent changes or an unreproducible behaviour?

**Comment by kumy** [comment]
First thought is that fix for #4780 might be a clue...

**Comment by Lineflyer** [comment]
It is no longer happening. Closing for now.

**Comment by Lineflyer** [comment]
@kumy 
I beleive there is still something wrong here:
I tested using 2015.06.03-RC1 version.

- I have GC3FJ5F stored in my lists
- I open it and go to the inventory
- I see one GK listed there named "17"

If I open the details of that GK the debug log shows a request to GK **and** GC although it should be clear that this is a GK not a GC-TB:

```
15:37:09.517 Debug cgeo 11463  [main] Loading GC3FJ5F Herminghauspark (11) from DB
15:37:14.657 Debug cgeo 11463  [network-5] GeokretyConnector.searchTrackable: gkid=32582
15:37:14.657 Debug cgeo 11463  [network-9] GET http://www.geocaching.com/track/details.aspx?tracker=GK7F46
15:37:14.657 Debug cgeo 11463  [network-5] GET http://api.geokretymap.org/export-details.php?gkid=32582
15:37:14.977 Debug cgeo 11463  [network-5] 200 (363 ms) GET http://api.geokretymap.org/export-details.php?gkid=32582
15:37:15.087 Debug cgeo 11463  [network-9] 200 (430 ms) GET http://www.geocaching.com/track/details.aspx?tracker=GK7F46
15:37:15.417 Warning cgeo 11463  [network-9] GCParser.searchTrackable: No trackable parsed
```

As this would mean, that we might send quite some requests to the wrong target with the new release its kind of blocking for the release...or not?

**Comment by kumy** [comment]
I hope I could try to have a look tonight

**Comment by Lineflyer** [comment]
Another good candidate for testing: GC1X9FR

**Comment by Lineflyer** [comment]
@Kumy: Did you find time to check?

**Comment by kumy** [comment]
Sorry wasn't possible. I'll expect tomorrow

**Comment by kumy** [comment]
If anyone would have a look, feel free to do so ;)

**Comment by kumy** [comment]
This comes from code here : https://github.com/cgeo/cgeo/blob/master/main/src/cgeo/geocaching/TrackableActivity.java#L270-L290

Comment say
> We query all the connectors that can handle the trackable in parallel as well as the local storage.
> We return the first positive result coming from a connector, or, if none, the result of loading from
> the local storage.

Then, `TravelbugConnector` has this pattern:
```
PATTERN_TB_CODE = Pattern.compile("(TB[0-9A-Z]+)|([0-9A-Z]{6})", Pattern.CASE_INSENSITIVE);
```

Effectively, GK id are matched by the second part of the pattern. The pattern is here to lookup TB by Tracking code.

Note: This remind me #4822, which talk about quite the same 'problem' (but it's not a duplicated bug).


@Lineflyer 
Having that said, is this really a bug?
Should we change something?

**Comment by Lineflyer** [comment]
If I understand the functionality correct, at least the pattern matching should IMHO be made different in case we look at the inventory of a cache. In this case the TB_CODE for a GC TB will **ALWAYS** start with TB while the GK will **always** start with GK as the inventory only holds the public code...or am I wrong?

I was only a little nervous as we will send more incorrect requests to GC after the release....but as this only happens if a user clicks on a GK it might still be not many occurences?!

**Comment by rsudev** [comment]
When we look at a cache inventory, the TB connector should already be known, as we have to actively retrieve GK for GC caches and never find GC TBs or the like outside a GC cache, right?

**Comment by kumy** [comment]
From what I've seen, the 'bad' request was not emited while looking at the cache inventory, but when opening the Trackable details.

**Comment by kumy** [comment]
Maybe we should pass some kind of "info" from the CacheDetailInventory (as an intent?) saying which brand of trackable we want to open, then limit "connector finding" to connector matching the brand.

This would limit the 'bad' request happening while opening a Trackable from cache inventory.

**Comment by kumy** [comment]
PR #5022 open.
Unit tests are still running here. (hope everything will be okay :p)

@rsudev are you still around? May you have a look please?

**Comment by rsudev** [comment]
@kumy Merged. Thanks!

**Comment by kumy** [comment]
Nice :)

**Comment by kumy** [comment]
How should this commit be applied to master?

**Comment by rsudev** [comment]
Already done :)

**Comment by Lineflyer** [comment]
Thanks @kumy.

Created a new RC version 2016.05.11-RC

**Comment by Lineflyer** [comment]
Retest of the scenario above with 2015.06.11-RC looks good.

**Comment by kumy** [comment]
Great :)
BTW, I'll be off until the end of the week...

**Comment by kumy** [comment]
This happens also while opening a `Trackable` from the `LogCacheActivity`.
